code-challenges.test.js

```diff
@@ -14,39 +14,146 @@
+
+//   ReferenceError: fruitWords is not defined
+
+// b) Create the function that makes the test pass.
+
+// GREEN:
+
+// PSUEDOCODE:
+// declare a function named fruitWords
+// input: array and string
+// output: array
+// Process: created a function fruitWords. passed arr and letter as parameters. filtered through arr with specified condition includes. once its filtered and finds specific letter return the full word in a array.
+// Reference link: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#:~:text=The%20filter()%20method%20is,included%20in%20the%20new%20array
 
+const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
 const letterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
 const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
-// b) Create the function that makes the test pass.
+const fruitWords = (arr, letter) => {
+    return arr.filter((arr) => arr.toLowerCase().includes(letter.toLowerCase()))
+}
+
+console.log(fruitWords(fruitArray, letterA))
+console.log(fruitWords(fruitArray, letterE))
+// PASS  ./code-challenges.test.js
+// fruitWords
+//   ✓ returns array of all words with particular letter (1 ms)
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+// RED:
+
+describe("allHands", () => {
+    it("determines whether or not array is a full house", () => {
+        const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+        const hand4 = [7, 2, 7, 2, 7]
+        // Expected output: true
+        expect(allHands(hand1)).toEqual(true)
+        expect(allHands(hand2)).toEqual(false)
+        expect(allHands(hand3)).toEqual(false)
+        expect(allHands(hand4)).toEqual(true)
+    })
+})
+// FAIL  ./code-challenges.test.js
+// allHands
+//   ✕ determines whether or not array is a full house
+
+// ● allHands › determines whether or not array is a full house
+
+//   ReferenceError: allHands is not defined
 
 // b) Create the function that makes the test pass.
+
+// GREEN:
+
+// PSUEDOCODE:
+// declare a function named allHands
+// input: array of numbers
+// output: boolean
+// Process:
+// *** Note: I utilized my resources (Google, classmates) and I was not able to fully grasp what was being explained to me about how certain code was written. So I was not able to write any code for this third challenge.
+
```

Comment:
Hey Dom - I wanted to give you some pseudo code that you can use to reference and try this problem again. If you need a better explanation, reach out and I can better explain or find a better resource.
Pseudo Code:
- Declare a function (allHands)
- The function (there are many different ways but the way we will tackle this will be conditionally) takes in an array to look for specific conditions (the ways that an array will meet a full house i.e. exactly one pair and one three of a kind).
- First 3 conditions: needs to see if the array index by index are strictly equal AND (logical &&) the next pair. If met it will return the boolean value.
- Else: it is not a full house